lib/itk/queue/publisher.ex

```diff
@@ -122,15 +122,17 @@ defmodule ITKQueue.Publisher do
           headers :: Headers.t(),
           stacktrace :: any,
           options :: Keyword.t()
-        ) :: no_return
+        ) :: pid()
   def publish_async(routing_key, message, headers, stacktrace, options)
       when is_bitstring(routing_key) do
     if testing?() do
       fake_publish(routing_key, message)
       :ok
     else
-      Task.async(fn ->
-        publish(routing_key, message, headers, stacktrace, options)
+      spawn(fn ->
```

Comment:
n00b question, why did we change `Task.async`? I didn't see somewhere using the `publish_async` function, but I'd expect any caller to `await` for the results instead of spawning it and forget about the results, what do you think?

Reply:
Yeah I don't see this function used anywhere either I think it's just code rot. It's not returning the result of publish anyway so a simple spawn works. We could probably remove `publish_async`.

Reply:
Ok, or maybe rename it so it's not confusing with the `async/await` style.